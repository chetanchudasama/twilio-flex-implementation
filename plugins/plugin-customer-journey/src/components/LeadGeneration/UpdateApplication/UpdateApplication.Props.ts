export interface OwnProps {
  updateApplication: () => void;
}

export type UpdateApplicationProps = OwnProps;
