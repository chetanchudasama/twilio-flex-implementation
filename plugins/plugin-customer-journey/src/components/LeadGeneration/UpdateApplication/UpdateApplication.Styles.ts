import styled from "react-emotion";
import { black, mediumGrey, secondary, white } from "@common/components";

export const UpdateApplicationStyles = styled("div")({
  background: white,
  padding: 24,
  marginLeft: -8,
  width: "100%",
  ".update-application-heading": {
    fontSize: 20,
    lineHeight: "24px",
    fontWeight: 600,
    color: black,
    paddingBottom: 8,
  },
  ".update-application-text": {
    fontSize: 14,
    paddingTop: 8,
  },
  ".update-application": {
    display: "flex",
    alignItems: "center",
  },
  ".update-button": {
    textAlign: "right",
  },
  ".update-application-btn": {
    margin: "5px 0",
    color: secondary,
    background: white,
    border: `1px solid ${mediumGrey}`,
    "&:hover": {
      backgroundColor: white,
    },
    borderRadius: 4,
    fontWeight: 600,
    fontSize: 14,
  },
});
