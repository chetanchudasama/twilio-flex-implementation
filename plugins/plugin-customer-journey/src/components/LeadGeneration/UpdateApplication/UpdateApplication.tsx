import React from "react";
import { Button, Grid } from "@material-ui/core";

import { UpdateApplicationStyles } from "./UpdateApplication.Styles";
import { UpdateApplicationProps } from "./UpdateApplication.Props";

export const UpdateApplication: React.FC<UpdateApplicationProps> = ({
  updateApplication,
}) => {
  return (
    <UpdateApplicationStyles>
      <Grid container spacing={16} className="update-application">
        <Grid item xs={8}>
          <Grid container>
            <Grid item xs={12} className="update-application-heading">
              Update application
            </Grid>
            <Grid item xs={12} className="update-application-text">
              This will save the above information to a note in this
              customer&apos;s application.
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={4} className="update-button">
          <Button
            variant="outlined"
            className="update-application-btn"
            onClick={updateApplication}
          >
            Update Application
          </Button>
        </Grid>
      </Grid>
    </UpdateApplicationStyles>
  );
};
