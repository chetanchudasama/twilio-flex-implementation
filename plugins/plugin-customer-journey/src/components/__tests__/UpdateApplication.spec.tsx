import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";

import { UpdateApplication } from "../LeadGeneration/UpdateApplication/UpdateApplication";
import { UpdateApplicationProps } from "../LeadGeneration/UpdateApplication/UpdateApplication.Props";

const props: UpdateApplicationProps = {
  updateApplication: jest.fn(),
};

describe("UpdateApplication", () => {
  it("renders correctly on load", () => {
    const wrapper = create(<UpdateApplication {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should call updateApplication method, when Update Application button is clicked", () => {
    const wrapper = mount(<UpdateApplication {...props} />);

    const updateApplicationButton = wrapper
      .find(".update-application-btn")
      .last();

    updateApplicationButton.simulate("click");

    expect(props.updateApplication).toHaveBeenCalled();
  });
});
