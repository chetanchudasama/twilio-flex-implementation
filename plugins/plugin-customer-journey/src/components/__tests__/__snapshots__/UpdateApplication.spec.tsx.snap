// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateApplication renders correctly on load 1`] = `
<div
  className="css-1j429wh"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24 update-application"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
    >
      <div
        className="MuiGrid-container-1"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 update-application-heading"
        >
          Update application
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 update-application-text"
        >
          This will save the above information to a note in this customer's application.
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-4-33 update-button"
    >
      <button
        className="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-outlined-106 update-application-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-99"
        >
          Update Application
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </button>
    </div>
  </div>
</div>
`;
